recipes/protobuf/all/conanfile.py

```diff
@@ -112,8 +116,13 @@ def _configure_cmake(self):
             cmake.definitions["protobuf_BUILD_LIBPROTOC"] = True
         if self._can_disable_rtti:
             cmake.definitions["protobuf_DISABLE_RTTI"] = not self.options.with_rtti
-        if self._is_msvc:
-            cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in msvc_runtime_flag(self)
+        if self._is_msvc or self._is_clang_cl:
+            runtime = msvc_runtime_flag(self)
+            if not runtime:
+                runtime = self.settings.get_safe("compiler.runtime")
+            cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in runtime
+        if self._is_clang_cl:
+            cmake.definitions["CMAKE_RC_COMPILER"] = os.environ.get("RC", "llvm-rc")
```

Comment:
Why is it needed? CMake bug? Lack of information in conan profile by a user?

Reply:
I see this fix: https://github.com/protocolbuffers/protobuf/pull/8139, it's part of protobuf >= 3.18.0. Maybe backport the patch in previous versions?

Reply:
okay, maybe it's worth to do, thanks for noticing

Reply:
Are going for the patch approach?

Reply:
It would require 8 patches (maybe less): 1 per version before 3.18.0
Otherwise, I've suggested https://github.com/conan-io/conan-center-index/pull/10005#discussion_r839409801 to not inject this trick if not required.